### PR TITLE
Add a remove button to the selections gallery

### PR DESCRIPTION
**Pull request by kf99916**
Make the selections gallery more powerful to support selections remove.

### CODE REVIEW

File: Source/SelectionsGallery/YPSelectionsGalleryCell.swift [hunk 1]
```diff
@@ -54,6 +54,20 @@ public class YPSelectionsGalleryCell: UICollectionViewCell {
...
     }
     
+    func addRemoveButton(target: Any?, action: Selector) {
+        let image = YPConfig.icons.removeImage
+        let removeButton = UIButton(frame: CGRect(origin: CGPoint.zero, size: image.size))
+        removeButton.setBackgroundImage(image, for: UIControl.State())
+        removeButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        self.addSubview(removeButton)
+        ["H:[subview]-12-|", "V:|-12-[subview]"].forEach { visualFormat in
```
Comment (NikKovIos) [inline Source/SelectionsGallery/YPSelectionsGalleryCell.swift:64]: Please change to https://github.com/freshOS/Stevia/issues for constraints here.

File: Source/SelectionsGallery/YPSelectionsGalleryVC.swift [hunk 2]
```diff
@@ -83,6 +94,11 @@ extension YPSelectionsGalleryVC: UICollectionViewDataSource {
             cell.imageView.image = video.thumbnail
             cell.setEditable(YPConfig.showsVideoTrimmer)
         }
+        
+        if !YPConfig.gallery.hidesRemoveButton {
+            cell.addRemoveButton(target: self, action: #selector(reomveButtonDidClick(sender:)))
```
Comment (NikKovIos) [inline Source/SelectionsGallery/YPSelectionsGalleryVC.swift]: Typo in `reomveButtonDidClick`

File: Source/Configuration/YPIcons.swift [hunk 3]
```diff
@@ -26,4 +26,5 @@ public struct YPIcons {
     public var captureVideoImage: UIImage = imageFromBundle("yp_iconVideoCapture")
     public var captureVideoOnImage: UIImage = imageFromBundle("yp_iconVideoCaptureRecording")
     public var playImage: UIImage = imageFromBundle("yp_play")
+    public var removeImage: UIImage = imageFromBundle("yp_remove")
```
Comment (NikKovIos) [inline Source/Configuration/YPIcons.swift:29]: Please change the filename of yp_remove from `Remove-universal-32@..png` to `yp_remove@..png`

File: Source/Configuration/YPImagePickerConfiguration.swift [hunk 4]
```diff
@@ -28,6 +28,9 @@ public struct YPImagePickerConfiguration {
     // Video configuration
     public var video = YPConfigVideo()
     
+    // Gallery configuration
+    public var gallery = YPConfigGallery()
```
Comment (NikKovIos) [inline Source/Configuration/YPImagePickerConfiguration.swift]: Nice👍

**Review by NikKovIos — CHANGES_REQUESTED** [review]

File: Source/Configuration/YPImagePickerConfiguration.swift [hunk 5]
```diff
@@ -217,6 +220,12 @@ public struct YPConfigVideo {
     public var trimmerMinDuration: Double = 3.0
 }
 
+/// Encapsulates gallery specific settings.
+public struct YPConfigGallery {
```
Comment (NikKovIos) [inline Source/Configuration/YPImagePickerConfiguration.swift]: Please rename to YPConfigSelectionsGallery

**Comment by kf99916** [comment]
@NikKovIos , the PR is refined according to your suggestion. Please help review them. Thanks.

**Comment by NikKovIos** [comment]
@kf99916 Still need one thing to change. (Layout with Stevia)

File: Source/SelectionsGallery/YPSelectionsGalleryCell.swift (hunk 1 above)
Comment (s4cha) [reply Source/SelectionsGallery/YPSelectionsGalleryCell.swift:64]: I'll take care of the changes :)

**Comment by s4cha** [comment]
This is a very cool feature cleanly implemented, what's more to ask ? 👏 🥇

**Pull request merged**